segmentstore/server/src/test/java/io/pravega/segmentstore/server/DataRecoveryTestUtils.java

```
@@ -0,0 +1,188 @@
+                            }
+                            return null;
+                        }).join();
+            }
+            for (TableKey k : segmentsInMD) {
+                String segmentName = k.getKey().toString();
+                log.info("Deleting segment : {} as it is not in storage", segmentName);
+                try {
+                    container.deleteStreamSegment(segmentName, TIMEOUT).join();
+                } catch (Throwable e) {
+                    log.error("Error while deleting the segment = {}", segmentName);
+                }
+            }
+            log.info("Recovery done for container = {}", containerId);
+        }
+    }
+
+    public static ArrayView getTableKey(String segmentName) {
+        return new ByteArraySegment(segmentName.getBytes(Charsets.UTF_8));
+    }
+
+    /**
+     * Deletes container-metadata segment and attribute index segment for the given container Id.
+     * @param tier2         Long term storage to delete the segments from.
+     * @param containerId   Id of the container for which the segments has to be deleted.
+     */
+    public static void deleteContainerMetadataSegments(Storage tier2, int containerId) {
+        deleteSegment(tier2, "_system/containers/metadata_" + containerId);
+        deleteSegment(tier2, "_system/containers/metadata_" + containerId + "$attributes.index");
+    }
+
+    /**
+     * Deletes the segment with given segment name from the given long term storage.
+     * @param tier2         Long term storage to delete the segment from.
+     * @param segmentName   Name of the segment to be deleted.
+     */
+    public static void deleteSegment(Storage tier2, String segmentName) {
+        try {
+            SegmentHandle segmentHandle = tier2.openWrite(segmentName).join();
+            tier2.delete(segmentHandle, TIMEOUT).join();
+        } catch (Throwable e) {
+            log.info("Error while deleting segment: {}", segmentName);
```

Let the exception bubble up

---

It makes the test fail all the times, if I let it bubble up. The reason is most of the time, container metadata segments for some containers don't exist while I am trying to delete them.

---

It's the only place where I am not letting the exception bubble up.